pkg/scalers/newrelic_scaler.go

```diff
@@ -0,0 +1,205 @@
+package scalers
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"strconv"
+
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	v2beta2 "k8s.io/api/autoscaling/v2beta2"
+	"k8s.io/apimachinery/pkg/labels"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/newrelic/newrelic-client-go/newrelic"
+	"github.com/newrelic/newrelic-client-go/pkg/nrdb"
+)
+
+const (
+	account    = "account"
+	queryKey   = "queryKey"
+	region     = "region"
+	metricName = "metricName"
+	nrql       = "nrql"
+	threshold  = "threshold"
+	noDataErr  = "noDataErr"
+)
+
+type newrelicScaler struct {
+	metadata *newrelicMetadata
+	nrClient *newrelic.NewRelic
+}
+
+type newrelicMetadata struct {
+	account     int
+	region      string
+	queryKey    string
+	metricName  string
+	noDataErr   bool
+	nrql        string
+	threshold   int
+	scalerIndex int
+}
+
+var newrelicLog = logf.Log.WithName("new-relic_scaler")
+
+func NewNewRelicScaler(config *ScalerConfig) (Scaler, error) {
+	meta, err := parseNewRelicMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing new-relic metadata: %s", err)
+	}
+
+	nrClient, err := newrelic.New(
+		newrelic.ConfigPersonalAPIKey(meta.queryKey),
+		newrelic.ConfigRegion(meta.region))
+
+	if err != nil {
+		log.Fatal("error initializing client:", err)
+	}
+
+	logMsg := fmt.Sprintf("Initializing New Relic Scaler (account %d in region %s)", meta.account, meta.region)
+
+	newrelicLog.Info(logMsg)
+
+	return &newrelicScaler{
+		metadata: meta,
+		nrClient: nrClient}, nil
+}
+
+func parseNewRelicMetadata(config *ScalerConfig) (*newrelicMetadata, error) {
+	meta := newrelicMetadata{}
+	var err error
+	if val, ok := config.TriggerMetadata[account]; ok && val != "" {
+		t, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %s: %s", account, err)
+		}
+		meta.account = t
+	} else {
+		return nil, fmt.Errorf("no %s given", account)
+	}
+
+	if val, ok := config.TriggerMetadata[nrql]; ok && val != "" {
+		meta.nrql = val
+	} else {
+		return nil, fmt.Errorf("no %s given", nrql)
+	}
+
+	meta.queryKey, err = GetFromAuthOrMeta(config, queryKey)
+	if err != nil {
+		return nil, fmt.Errorf("no %s given", queryKey)
+	}
+
+	if val, ok := config.TriggerMetadata[metricName]; ok && val != "" {
```

Comment:
I think we can drop this one, see following discussion:
https://github.com/kedacore/keda/pull/2354#discussion_r758090060

Reply:
@zroubalik I don't understand what you want to be dropped from here.